### PR TITLE
first stab at calling ise driver summary batch.. unit tests to come

### PR DESCRIPTION
**Pull request by dehang**

### CODE REVIEW

File: e2e-tests/__tests__/get-driver-summary.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  };
+  const config: ApiConfig = {
+    host: 'compliance-svc-qa.connectedfleet.io',
+    credentials: 'c3ZjX29pX2lzZV9iYXNpYzoycHM2ZERGUlI4OEpoakY4',
+    logLevel: 'debug'
+  };
+
+  const result = getDriverSummaryBatch('5232', [driver], config);
+
+  result.then(d => console.log(d)); // TODO not sure why this logs the days array as '[Object]'
```
Comment (dehang) [inline e2e-tests/__tests__/get-driver-summary.ts]: see comment here.

File: src/__tests__/__snapshots__/index.ts.snap [hunk 2]
```diff
@@ -1,3 +0,0 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (dehang) [inline src/__tests__/__snapshots__/index.ts.snap]: Had to run the -u command to remove the snapshot file. Not sure what these files are for, seems like they're for UI testing?

File: src/get-driver-summary.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      if (res.ok) {
+        return res.json();
+      }
+      const err = `Non 200 returned from driver summary batch endpoint, status: ${
+        res.status
+      }, response: ${res.statusText}`;
+      log.error(err);
+      throw new Error(err);
+    })
+    .then(response => response as IDriverSummaryResponse[]); // TODO: couldn't get this to return above
```
Comment (dehang) [inline src/get-driver-summary.ts]: see comment here.

File: e2e-tests/__tests__/get-driver-summary.ts (hunk 1 above)
Comment (DSchau) [reply e2e-tests/__tests__/get-driver-summary.ts]: Try `console.log(JSON.stringify(d, null, 2))`

File: package.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
     "typescript": "~2.6.2"
   },
   "dependencies": {
-    "@peoplenet/ise-common-util": "~0.0.4",
+    "@peoplenet/ise-common-util": "0.0.4",
```
Comment (DSchau) [inline package.json]: Not sure how this got in here, but we should prefer tilde `~`, which basically means get any version in this range.

File: src/__tests__/__snapshots__/index.ts.snap (hunk 2 above)
Comment (DSchau) [reply src/__tests__/__snapshots__/index.ts.snap]: A snapshot is basically metadata that is diffed against. If it changes, the test will fail!

But yeah, go ahead and run `-u` if a diff is outdated or needs updating. In the PR we'll see that a diff changed, and we'll have to gauge whether or not that change was _supposed_ to have happened

File: src/__tests__/index.ts [hunk 5]
```diff
@@ -1,5 +1,7 @@
-import * as SDK from '..';
+jest.mock('node-fetch');
+import fetch, { Response } from 'node-fetch';
+import { IApiConfig } from '../interfaces';
 
-test('it has expected exports', () => {
-  expect(Object.keys(SDK)).toMatchSnapshot();
+test('in progress', () => {
+  expect(1).toEqual(1);
```
Comment (DSchau) [inline src/__tests__/index.ts]: 🎉

Comment (DSchau) [reply src/__tests__/index.ts]: Next up, `expect(1 + 1).toEqual(2)`

File: src/get-driver-summary.ts (hunk 3 above)
Comment (DSchau) [reply src/get-driver-summary.ts]: What does `couldn't get this to return above` exactly mean?

Comment (dehang) [reply src/get-driver-summary.ts]: as in couldn't get it to return from res.json() but maybe it needs to happen in the second then clause?

File: src/__tests__/index.ts (hunk 5 above)
Comment (dehang) [reply src/__tests__/index.ts]: i don't know man.. that's hard

File: e2e-tests/__tests__/get-driver-summary.ts (hunk 1 above)
Comment (dehang) [reply e2e-tests/__tests__/get-driver-summary.ts]: 🗡

**Pull request merged**